src/dilpack/clustering/spectral.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+from sklearn.neighbors import radius_neighbors_graph
+
+
+class SpectralClustering:
+    def __init__(self, X, num_clusters):
+        self.K = num_clusters
+        self.max_iterations = 1000
+        self.num_examples = X.shape[0]
+        self.num_features = X.shape[0]
+
+    # Get eigenvectors and eigenvalues
+    def get_eigs_from_laplacian(self, X):
```

Comment:
Is the X here different from the one in __init__? If they are the same the it might make sense to do a self.X instead.

Reply:
that makes sense, I can make that change now